### PR TITLE
refactor build.gradle

### PR DESCRIPTION
**Pull request by sejoDjoseon**
upgrade kustomer android 2.9.+

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,65 +1,41 @@
...
-def DEFAULT_COMPILE_SDK_VERSION = 28
-def DEFAULT_BUILD_TOOLS_VERSION = '28.0.3'
+def DEFAULT_COMPILE_SDK_VERSION = 29
+def DEFAULT_BUILD_TOOLS_VERSION = '29.0.0'
 def DEFAULT_MIN_SDK_VERSION = 21
-def DEFAULT_TARGET_SDK_VERSION = 28
-def LIFECYLCE_VERSION = "2.3.1"
+def DEFAULT_TARGET_SDK_VERSION = 29
+def DEFAULT_GRADLE_VERSION= "4.1.2"
+def DEFAULT_KOTLIN_VERSION= "1.4.21"
```
Comment (pauleta6) [inline android/build.gradle]: they are already declared in the gradle.properties file

File: android/build.gradle [hunk 2]
```diff
@@ -1,65 +1,41 @@
...
+        rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+    }
+    
+    ext {
+        kotlinVersion = '1.4.21'
+        gradleVersion = '4.1.2'
+        buildToolsVersion = "29.0.0"
+        minSdkVersion = 21
+        compileSdkVersion = 29
+        targetSdkVersion = 29
```
Comment (pauleta6) [inline android/build.gradle:16]: why don't you use the ones created on top?

File: android/build.gradle [hunk 3]
```diff
@@ -1,65 +1,41 @@
...
+        google()
+        mavenCentral()
+    }
 
-        dependencies {
-            classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.50")
-        }
+    dependencies {
+        classpath "com.android.tools.build:gradle:4.1.2"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.21"
```
Comment (pauleta6) [inline android/build.gradle]: same here

File: android/build.gradle (hunk 1 above)
Comment (sejoDjoseon) [reply android/build.gradle]: treu

**Review by pauleta6 — APPROVED** [review]

**Pull request merged**